### PR TITLE
97 actions take time

### LINKED ISSUE #97: Make actions take time
- [x] Harvesting resources should take time.
- [x] Building structures should take time.

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
zDNPTyX7cN

Actions can now take time to perform using `AbstractTimeAction`.

Two Actions have been refactored to take time:
- `HarvestAction`
- `BuildAction`

Structures being built by multiple beings at the same time will be built faster, proportionally to how many builders are working on it.

## Test Coverage
image
image
image

## Diagrams
com thebois models beings actions
com thebois models world resources

Closes #97

### CODE REVIEW

File: src/main/java/com/thebois/models/beings/actions/AbstractTimeAction.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package com.thebois.models.beings.actions;
+
+import java.io.Serializable;
+
+import com.thebois.models.beings.IActionPerformer;
+
```
Comment (jonathan-lindqvist) [inline src/main/java/com/thebois/models/beings/actions/AbstractTimeAction.java:6]: Might as well add author while you create a whole new file 😄 👍

File: src/main/java/com/thebois/models/beings/actions/IAction.java [hunk 2]
```diff
@@ -8,11 +8,12 @@
 public interface IAction {
 
     /**
-     * Performs the task on a given performer.
```
Comment (jonathan-lindqvist) [inline src/main/java/com/thebois/models/beings/actions/IAction.java:11]: bild

**Review by jonathan-lindqvist — APPROVED** [review]
Starting too feel like a game 👀

**Review by Matte02 — APPROVED** [review]
Lgtm!

**Pull request merged**